# Исследование надежности заемщиков.

### Шаг 1. Обзор данных

In [ ]:
import pandas as pd

In [ ]:
data = pd.read_csv('data.csv')
data.head(19)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450  строительство собственной недвижимости  
16  289202.704229                            недвижимость  
17   90410.586745              строительст

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


### Шаг 2.1 Заполнение пропусков

In [ ]:
data[data['total_income'].isna()]

children  days_employed  dob_years education  education_id  \
12            0            NaN         65   среднее             1   
26            0            NaN         41   среднее             1   
29            0            NaN         63   среднее             1   
41            0            NaN         50   среднее             1   
55            0            NaN         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            NaN         47   Среднее             1   
21495         1            NaN         50   среднее             1   
21497         0            NaN         48    ВЫСШЕЕ             0   
21502         1            NaN         42   среднее             1   
21510         2            NaN         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              NaN                   сыграть свадьбу  
26              NaN                       образование  
29              NaN  строительство жилой недвижимости  
41              NaN  сделка с подержанным автомобилем  
55              NaN                   сыграть свадьбу  
...             ...                               ...  
21489           NaN              сделка с автомобилем  
21495           NaN                           свадьба  
21497           NaN        строительство недвижимости  
21502           NaN  строительство жилой недвижимости  
21510           NaN           приобретение автомобиля  

[2174 rows x 12 columns]

In [ ]:
21525 - 19351

2174

In [ ]:
2174/21525 * 100

10.099883855981417

Помимо столбца 'days_employed' пропуски присутствуют в столбце 'total_income', получается у нас нет данных по общему доходу у 2174 человек, что составляет 10.1% от всех данных в каждом из двух столбцов. Почему могли появиться эти пропуки?
Пропуски могли появиться по техническим причинам, человеческий фактор, так же так как пропуски в каждом из двух столбцах находяться в одинаковых строчках, то можно увидить какую нибудь взаимосвязь, может во время опроса если ты не указываешь дней на работе, то и общий доход не имеет значения. Так же есть шанс что некоторые люди в целях конфидециальности намерренно пропускали и не заполняли эти данные, так же есть предположение что для пенсионеров/студентов/безработных эти данные не должны указываться. Поэтому заполним эти пропуски медианными значениями, потому что они наногокорректней подойдут, чем то же среднее значение, потому что могут присутствовать выбросы, которые испортят среднее значение.

In [ ]:
total_income_median = data['total_income'].median()
total_income_median

145017.93753253992

In [ ]:
data['total_income'] = data['total_income'].fillna(total_income_median)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      21525 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      21525 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


### Шаг 2.2 Проверка данных на аномалии и исправления.

In [ ]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

Аномалии могут возникнуть так же из за технических проблем, из за человеческого фактора, так же может быть в столбце 'days_employed' аномальные данные могли возникнуть из за времени исчесления, кто то мог записать в годах, кто то в неделях, кто то в чаасах. Абсолютно точно не может быть отрицательного стажа работы, заменим его на модуль этого числа(предположим что минус появился по техническим причинам во время переноса данных), так же не может быыть слишком больших данных даже при учете, что человек уже пенсионер, а на пенсию выходят в 60 лет и даже если он где то переработал и так далее возьмем максимальный стаж работы: 50 лет, а это 18250 дней,и все данные что больше заменим на это значение

In [ ]:
50*365

18250

In [ ]:
data['days_employed'] = abs(data['days_employed'])
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   Среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

In [ ]:
data.loc[data['days_employed']>18250, 'days_employed'] = 18250
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   Среднее             1   
3         3    4124.747207         32   среднее             1   
4         0   18250.000000         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

In [ ]:
data.loc[data['days_employed'] == 18250.000000]

children  days_employed  dob_years education  education_id  \
4             0        18250.0         53   среднее             1   
18            0        18250.0         53   среднее             1   
24            1        18250.0         57   среднее             1   
25            0        18250.0         67   среднее             1   
30            1        18250.0         62   среднее             1   
...         ...            ...        ...       ...           ...   
21505         0        18250.0         53   среднее             1   
21508         0        18250.0         62   среднее             1   
21509         0        18250.0         59    высшее             0   
21518         0        18250.0         59   СРЕДНЕЕ             1   
21521         0        18250.0         67   среднее             1   

               family_status  family_status_id gender income_type  debt  \
4           гражданский брак                 1      F   пенсионер     0   
18            вдовец / вдова                 2      F   пенсионер     0   
24     Не женат / не замужем                 4      F   пенсионер     0   
25           женат / замужем                 0      M   пенсионер     0   
30           женат / замужем                 0      F   пенсионер     0   
...                      ...               ...    ...         ...   ...   
21505       гражданский брак                 1      M   пенсионер     0   
21508        женат / замужем                 0      M   пенсионер     0   
21509        женат / замужем                 0      M   пенсионер     0   
21518        женат / замужем                 0      F   пенсионер     0   
21521        женат / замужем                 0      F   пенсионер     0   

        total_income                                purpose  
4      158616.077870                        сыграть свадьбу  
18      56823.777243     на покупку подержанного автомобиля  
24     290547.235997  операции с коммерческой недвижимостью  
25      55112.757732                   покупка недвижимости  
30     171456.067993  операции с коммерческой недвижимостью  
...              ...                                    ...  
21505   75439.993167                        сыграть свадьбу  
21508   72638.590915                           недвижимость  
21509   73029.059379               операции с недвижимостью  
21518  153864.650328                   сделка с автомобилем  
21521  155999.806512                   сделка с автомобилем  

[3446 rows x 12 columns]

In [ ]:
days_employed_median = data['days_employed'].median()
days_employed_median

2194.220566878695

In [ ]:
data['days_employed'] = data['days_employed'].fillna(days_employed_median)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     21525 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      21525 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


### Шаг 2.3. Изменение типов данных.

In [ ]:
data['total_income'] = data['total_income'].astype('int')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     21525 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      21525 non-null  int64  
 11  purpose           21525 non-null  object 
dtypes: float64(1), int64(6), object(5)
memory usage: 2.0+ MB


### Шаг 2.4. Удаление дубликатов.

Для начала удалю все явные дубликаты

In [ ]:
data.shape

(21525, 12)

In [ ]:
data = data.drop_duplicates()
data.shape

(21471, 12)

In [ ]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   Среднее             1   
3         3    4124.747207         32   среднее             1   
4         0   18250.000000         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

In [ ]:
data['education'].unique()

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

In [ ]:
data.loc[:, 'education'] = data['education'].str.lower()

In [ ]:
data = data.drop_duplicates()
data.shape

(21454, 12)

In [ ]:
data.loc[:, 'family_status'] = data['family_status'].str.lower()

In [ ]:
data.loc[:, 'income_type'] = data['income_type'].str.lower()

In [ ]:
data.loc[:, 'purpose'] = data['purpose'].str.lower()

In [ ]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

Не может быть отрицательгное количсетво детей и не может быть 20 это слишком много детей, будем считать, что максимум 5 детей

In [ ]:
data['children'] = abs(data['children'])

In [ ]:
data['children'].unique()

array([ 1,  0,  3,  2,  4, 20,  5])

In [ ]:
data.loc[data['children'] > 5, 'children'] = 5

In [ ]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

In [ ]:
data['gender'].unique()

array(['F', 'M', 'XNA'], dtype=object)

Заменила значение 'XNA' в столбце 'gender' на 'F' :

In [ ]:
data.loc[data['gender'] == 'XNA', 'gender'] = 'F'

In [ ]:
data['gender'].unique()

array(['F', 'M'], dtype=object)

Обнаружила неявные дубликаты, распределила все цели получения кредита по четырем категориям и заменила названия дублирующие один и тот же смысл

In [ ]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

In [ ]:
data = data.drop_duplicates()
data.shape

(21454, 12)

In [ ]:
house = ['покупка жилья', 'операции с жильем', 'покупка жилья для семьи', 'покупка недвижимости', 'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'покупка своего жилья',
       'операции с недвижимостью', 'покупка жилья для сдачи', 'ремонт жилью', 'покупка коммерческой недвижимости', 'покупка жилой недвижимости']

In [ ]:
avto = ['приобретение автомобиля', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля', 'автомобили', 'сделка с подержанным автомобилем', 'автомобиль', 'свой автомобиль', 'сделка с автомобилем', 'на покупку автомобиля']

In [ ]:
edu = ['дополнительное образование', 'образование', 'заняться образованием', 'получение образования', 'получение дополнительного образования', 'получение высшего образования', 'профильное образование', 'высшее образование', 'заняться высшим образованием']

In [ ]:
wadding = ['сыграть свадьбу', 'на проведение свадьбы', 'свадьба']

In [ ]:
for i in range(len(house)):
  data.loc[data['purpose'] == house[i], 'purpose'] = 'жилье'

In [ ]:
for i in range(len(avto)):
  data.loc[data['purpose'] == avto[i], 'purpose'] = 'автомобиль'

In [ ]:
for i in range(len(edu)):
  data.loc[data['purpose'] == edu[i], 'purpose'] = 'образование'

In [ ]:
for i in range(len(wadding)):
  data.loc[data['purpose'] == wadding[i], 'purpose'] = 'свадьба'

In [ ]:
data['purpose'].unique()

array(['жилье', 'автомобиль', 'образование', 'свадьба'], dtype=object)

In [ ]:
data = data.drop_duplicates()
data.shape

(21120, 12)

In [ ]:
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0   18250.000000         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

       purpose  
0        жилье  
1   автомобиль  
2        жилье  
3  образование  
4      свадьба

### Шаг 2.5. Формирование дополнительных датафреймов словарей, декомпозиция исходного датафрейма.

In [ ]:
data['education'].unique()

array(['высшее', 'среднее', 'неоконченное высшее', 'начальное',
       'ученая степень'], dtype=object)

In [ ]:
data['education_id'].unique()

array([0, 1, 2, 3, 4])

In [ ]:
data_e = [['высшее', 0], ['среднее', 1], ['неоконченное высшее', 2], ['начальное', 3], ['ученая степень', 4]]
col_e = ['education', 'education_id']

In [ ]:
education_data = pd.DataFrame(data=data_e, columns=col_e)
education_data

education  education_id
0               высшее             0
1              среднее             1
2  неоконченное высшее             2
3            начальное             3
4       ученая степень             4

In [ ]:
data['family_status'].unique()

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'не женат / не замужем'], dtype=object)

In [ ]:
data['family_status_id'].unique()

array([0, 1, 2, 3, 4])

In [ ]:
data_f = [['женат / замужем', 0], ['гражданский брак', 1], ['вдовец / вдова', 2], ['в разводе', 3], ['не женат / не замужем', 4]]
col_f = ['family_status', 'family_status_id']

In [ ]:
family_status_data = pd.DataFrame(data=data_f, columns=col_f)
family_status_data

family_status  family_status_id
0        женат / замужем                 0
1       гражданский брак                 1
2         вдовец / вдова                 2
3              в разводе                 3
4  не женат / не замужем                 4

In [ ]:
data = data.drop('education', axis=1)
data.head()

children  days_employed  dob_years  education_id     family_status  \
0         1    8437.673028         42             0   женат / замужем   
1         1    4024.803754         36             1   женат / замужем   
2         0    5623.422610         33             1   женат / замужем   
3         3    4124.747207         32             1   женат / замужем   
4         0   18250.000000         53             1  гражданский брак   

   family_status_id gender income_type  debt  total_income      purpose  
0                 0      F   сотрудник     0        253875        жилье  
1                 0      F   сотрудник     0        112080   автомобиль  
2                 0      M   сотрудник     0        145885        жилье  
3                 0      M   сотрудник     0        267628  образование  
4                 1      F   пенсионер     0        158616      свадьба

In [ ]:
data = data.drop('family_status', axis=1)
data.head()

children  days_employed  dob_years  education_id  family_status_id gender  \
0         1    8437.673028         42             0                 0      F   
1         1    4024.803754         36             1                 0      F   
2         0    5623.422610         33             1                 0      M   
3         3    4124.747207         32             1                 0      M   
4         0   18250.000000         53             1                 1      F   

  income_type  debt  total_income      purpose  
0   сотрудник     0        253875        жилье  
1   сотрудник     0        112080   автомобиль  
2   сотрудник     0        145885        жилье  
3   сотрудник     0        267628  образование  
4   пенсионер     0        158616      свадьба

### Шаг 2.6. Категоризация дохода.

In [ ]:
data.loc[data['total_income'] <= 30000,'total_income_category'] = 'E'
data.loc[(data['total_income'] >= 30001) & (data['total_income'] <= 50000),'total_income_category'] = 'D'
data.loc[(data['total_income'] >= 50001) & (data['total_income'] <= 200000),'total_income_category'] = 'C'
data.loc[(data['total_income'] >= 200001) & (data['total_income'] <= 1000000),'total_income_category'] = 'B'
data.loc[data['total_income'] >= 1000001,'total_income_category'] = 'A'
data.tail()

children  days_employed  dob_years  education_id  family_status_id  \
21520         1    4529.316663         43             1                 1   
21521         0   18250.000000         67             1                 0   
21522         1    2113.346888         38             1                 1   
21523         3    3112.481705         38             1                 0   
21524         2    1984.507589         40             1                 0   

      gender income_type  debt  total_income     purpose total_income_category  
21520      F   компаньон     0        224791       жилье                     B  
21521      F   пенсионер     0        155999  автомобиль                     C  
21522      M   сотрудник     1         89672       жилье                     C  
21523      M   сотрудник     1        244093  автомобиль                     B  
21524      F   сотрудник     0         82047  автомобиль                     C

### Шаг 2.7. Категоризация целей кредита.

ну это уе до сделала

### Ответы на вопросы.

##### Вопрос 1:

Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [ ]:
children_debt = data.groupby('children')['debt'].mean()
children_debt

children
0    0.076723
1    0.092747
2    0.095145
3    0.082067
4    0.100000
5    0.094118
Name: debt, dtype: float64

##### Вывод 1:

Нет четкой зависимости: чем больше детей — тем выше риск

##### Вопрос 2:

Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [ ]:
family_status_debt = data.groupby('family_status_id')['debt'].mean()
family_status_debt

family_status_id
0    0.076929
1    0.094083
2    0.066808
3    0.071249
4    0.098420
Name: debt, dtype: float64

In [ ]:
family_status_data

family_status  family_status_id
0        женат / замужем                 0
1       гражданский брак                 1
2         вдовец / вдова                 2
3              в разводе                 3
4  не женат / не замужем                 4

##### Вывод 2:

Минимальный риск у вдовцов/вдов (6.68%) и людей в разводе (7.12%)
Максимальный риск у неженатых/незамужних (9.84%) и людей в гражданском браке (9.41%)
женат/замужем имеют средний риск (7.69%)

##### Вопрос 3:

Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [ ]:
total_income_debt = data.groupby('total_income_category')['debt'].mean()
total_income_debt

total_income_category
A    0.080000
B    0.070621
C    0.086596
D    0.060000
E    0.090909
Name: debt, dtype: float64

##### Вывод 3:

Самый низкий риск у клиентов с доходом 30 001–50 000
стабильный средний доходом,  небольшие кредиты
Самый высокий риск у клиентов с доходом 0–30 000
Низкий доход, финансовые трудности
Клиенты с доходом A имеют риск 8.00%, что выше, чем у группы D
Это может быть связано с крупными кредитами, которые сложнее погашать

##### Вопрос 4:

Как разные цели кредита влияют на его возврат в срок?


In [ ]:
purpose_debt = data.groupby('purpose')['debt'].mean()
purpose_debt

purpose
автомобиль     0.094101
жилье          0.073832
образование    0.093340
свадьба        0.080659
Name: debt, dtype: float64

##### Вывод 4:

Минимальный риск у кредитов на жилье.
Это связано с тем, что:
Жилье это залоговое имущество => снижает риски для банка.
длительный срок => уменьшает ежемесячную нагрузку.
Максимальный риск у кредитов на автомобиль (9.41%) и образование (9.33%).
Причины:Автомобиль быстро теряет в цене.
Образование не гарантирует немедленного роста дохода => не гарантирует погашение.
Свадьба (8.07%) — средний риск.

## Общий вывод:

In [ ]:
all_debt = data.groupby(['purpose', 'total_income_category', 'family_status_id'])['debt'].mean().reset_index()
all_debt

purpose total_income_category  family_status_id      debt
0   автомобиль                     A                 0  0.000000
1   автомобиль                     A                 4  0.000000
2   автомобиль                     B                 0  0.082232
3   автомобиль                     B                 1  0.085714
4   автомобиль                     B                 2  0.066667
..         ...                   ...               ...       ...
58     свадьба                     A                 1  0.000000
59     свадьба                     B                 1  0.051376
60     свадьба                     C                 1  0.090064
61     свадьба                     D                 1  0.088235
62     свадьба                     E                 1  0.000000

[63 rows x 4 columns]

Ключевые риски : Низкий доход, кредиты на автомобили/образование, неженатые клиенты.

Важно оценивать заемщиков комплексно (доход + семейный статус + цель кредита).

Данные подтверждают, что социально-демографические факторы и цели кредита влияют на возврат займа, но их анализ требует учета взаимосвязей.




